include/thread_pool/thread_pool.hpp

```diff
@@ -20,10 +21,31 @@ namespace tp
 
 template <typename Task, template<typename> class Queue>
 class ThreadPoolImpl;
-using ThreadPool = ThreadPoolImpl<FixedFunction<void(), 128>,
-                                  MPMCBoundedQueue>;
+using NonBlockingThreadPool = ThreadPoolImpl<FixedFunction<void(), 128>, MPMCBoundedQueue>;
+
+template<class T>
+class BlockingQueue {
+
+public:
+    BlockingQueue(size_t size) :
+        _decorableQueue(size)
+    {}
+
+    bool pop(T& item) {
+        _decorableQueue.timeout_move_pop(item, 100000);
```

Comment:
The timeout should be made into a parameters passed to the queue. Also it might make more sense to move this class to a different file.

Reply:
Done, although the timeout cannot be dynamically configured as per the current template structure of thread pool